src/widgets.rs

```diff
@@ -561,4 +607,78 @@ mod tests {
         widget.render_ref(buf.area, &mut buf);
         assert_eq!(buf, Buffer::with_lines(["                    "]));
     }
+
+    fn single_line_buffer() -> Buffer {
+        Buffer::empty(Rect::new(0, 0, 20, 1))
+    }
+
+    #[rstest]
+    fn str_render() {
+        let mut buf = single_line_buffer();
+        "hello world".render(buf.area, &mut buf);
+        assert_eq!(
+            buf,
+            Buffer::with_lines(["hello world         "]),
+            "str.render()"
+        );
```

Comment:
nit - with this automatically formatting to 5 lines for the assertion, this would probably be better with actual methods (I think you end up with 1 less line).

Reply:
can you clarify what you mean by "with actual methods"?

Reply:
https://github.com/ratatui-org/ratatui/pull/952/commits/d430470d0c8fe8f26f19f60c2b1058c608d9307d